packages/@pollyjs/adapter/src/index.js

```diff
@@ -176,8 +152,23 @@ export default class Adapter {
     if (recordingEntry) {
       await pollyRequest._emit('beforeReplay', recordingEntry);
 
-      if (this.shouldReRecord(pollyRequest, recordingEntry)) {
-        return this.record(pollyRequest);
+      if (isExpired(recordingEntry.startedDateTime, config.expiresIn)) {
+        const message =
+          'Recording for the following request has expired.\n' +
+          `${recordingEntry.request.method} ${recordingEntry.request.url}\n`;
+
+        if (config.expiryStrategy === EXPIRY_STRATEGIES.ERROR) {
+          this.assert(message);
+        } else if (config.expiryStrategy === EXPIRY_STRATEGIES.WARN) {
+          console.warn(`[Polly] ${message}`, recordingEntry);
+        } else if (config.expiryStrategy === EXPIRY_STRATEGIES.RECORD) {
+          return this.record(pollyRequest);
+        } else {
+          // TODO should there be a generic validator available for all config options?
```

Comment:
Good call out and makes sense to me. Could be a future PR since it's beyond the scope of this PR